Framework/PythonInterface/plugins/algorithms/LoadElementalAnalysisData.py

```diff
@@ -0,0 +1,177 @@
+num_files_per_detector = 3
+
+
+class LoadElementalAnalysisData(PythonAlgorithm):
+    def category(self):
+        """
+        Returns category
+        """
+        return "DataHandling"
+
+    def PyInit(self):
+        self.declareProperty(name='Run', defaultValue=0, validator=IntBoundedValidator(lower=0),
+                             doc='Run number to load')
+
+        self.declareProperty(WorkspaceGroupProperty(name='GroupWorkspace', defaultValue='',
+                                                    direction=Direction.Output),
+                             doc='Output group workspace for run')
+
+    def validateInputs(self):
+        issues = dict()
+        checkRun = self.search_user_dirs()
+        if not checkRun:
+            issues['Run'] = "Cannot find files for run " + self.getPropertyValue("Run")
+        return issues
+
+    def PyExec(self):
+        run = self.getPropertyValue("Run")
+        to_load = self.search_user_dirs()
+        workspaces = {
+            filename: self.get_filename(filename, run)
+            for filename in to_load if self.get_filename(filename, run) is not None
+        }
+        unique_workspaces = {}
+        for path, workspace in workspaces.items():
+            if workspace not in unique_workspaces.values():
+                unique_workspaces[path] = workspace
+        workspaces = unique_workspaces
+
+        self.format_workspace(workspaces)
+        self.merge_workspaces(run, workspaces.values())
+
+    def pad_run(self):
+        """ Pads run number: i.e. 123 -> 00123; 2695 -> 02695 """
+        return str(self.getProperty("Run").value).zfill(5)
+
+    def search_user_dirs(self):
+        """ Finds all files for the run number provided """
+        files = []
+        for user_dir in config["datasearch.directories"].split(";"):
+            path = os.path.join(user_dir, "ral{}.rooth*.dat".format(self.pad_run()))
+            files.extend([file for file in glob.iglob(path)])
+        return files
+
+    def get_filename(self, path, run):
+        """
+        Returns the overall workspace name
+        """
+        try:
+            detectors_num = str(int(path.rsplit(".", 2)[1][5]) - 1)
+            run_type = type_keys[path.rsplit(".")[1][-2:]]
+            return "_".join([detectors_num, run_type, str(run)])
+        except KeyError:
+            return None
+
+    def format_workspace(self, inputs):
+        """ inputs is a dict mapping filepaths to output names """
+        for path, output in inputs.items():
+            workspace = LoadAscii(path, OutputWorkspace=output)
+            workspace.getAxis(0).setUnit("Label").setLabel("Energy", "keV")
+
+    def merge_workspaces(self, run, workspaces):
+        """ where workspaces is a tuple of form:
+                (filepath, ws name)
+        """
+        # detectors is a dictionary of {detector_name : [names_of_workspaces]}
+        detectors = {f"{run}; Detector {x}": [] for x in range(1, 5)}
```

Comment:
instead of naming the workspaces as `run_detector`, we should use the same name as the group workspace instead of the run. Does that make sense? 

The user can then select it to have a more meaningful name e.g. 12BC_silver_coin_1